Description: Make Lcalc (1.23) build with GCC 5 (remove default parameters from definitions)
 GCC 5.1 (i.e., g++) again got stricter with respect to C++ conformance,
 and hence another piece of the changes necessary to let Lcalc build with
 Clang (cf. Sage trac #12437) is needed for g++ as well, namely forbidden
 default parameters in function definitions ("Part II").
Author: Leif Leonhardy <not.really@online.de>
Origin: http://git.sagemath.org/sage.git/tree/build/pkgs/lcalc/patches/lcalc-1.23_default_parameters_2.patch
Bug: http://trac.sagemath.org/ticket/18316
Bug-Debian: http://bugs.debian.org/777932

--- a/include/Lgamma.h
+++ b/include/Lgamma.h
@@ -77,7 +77,7 @@
 //n=0 should just give log_GAMMA(z)... thus making log_GAMMA
 //code obsolete. But leave log_GAMMA intact anyways.
 template <class ttype>
-precise(ttype) log_GAMMA (ttype z,int n=0)
+precise(ttype) log_GAMMA (ttype z,int n)
 {
     int M;
     precise(ttype) log_G,r,r2,y;
@@ -230,7 +230,7 @@
 //value exp_w which holds exp(-w)
 //computes G(z,w), so there's an extra w^(-z) factor.
 template <class ttype>
-Complex inc_GAMMA (ttype z,ttype w, const char *method="temme", ttype exp_w = 0, bool recycle=false)
+Complex inc_GAMMA (ttype z,ttype w, const char *method, ttype exp_w, bool recycle)
 {
 
     Complex G;
@@ -334,7 +334,7 @@
 
 
 template <class ttype>
-ttype cfrac_GAMMA (ttype z,ttype w, ttype exp_w=0, bool recycle=false)  //computes G(z,w) via continued fraction
+ttype cfrac_GAMMA (ttype z,ttype w, ttype exp_w, bool recycle)  //computes G(z,w) via continued fraction
 {
 
         ttype G;
@@ -424,7 +424,7 @@
 }
 
 template <class ttype>
-ttype asympt_GAMMA (ttype z,ttype w, ttype exp_w = 0, bool recycle=false)  //computes G(z,w) via asymptotic series
+ttype asympt_GAMMA (ttype z,ttype w, ttype exp_w, bool recycle)  //computes G(z,w) via asymptotic series
 {
 
         if(my_verbose>3) cout << "called asympt_GAMMA("<<z<<","<<w<<")"<< endl;
@@ -446,7 +446,7 @@
 
 
 template <class ttype>
-ttype comp_inc_GAMMA (ttype z,ttype w,ttype exp_w = 0, bool recycle=false)  //computes g(z,w)
+ttype comp_inc_GAMMA (ttype z,ttype w,ttype exp_w, bool recycle)  //computes g(z,w)
 {
 
     ttype g;
@@ -604,7 +604,7 @@
 }
 
 template <class ttype>
-Complex gamma_sum(Complex s, int what_type, ttype *coeff, int N, Double g, Complex l, Double Q, Long Period, Complex delta=1, const char *method="temme")
+Complex gamma_sum(Complex s, int what_type, ttype *coeff, int N, Double g, Complex l, Double Q, Long Period, Complex delta, const char *method)
 {
     Complex SUM=0;
 
